### PR TITLE
[WFLY-14682] Upgrade ironjacamar to 1.4.31.Final 385a118

### PR DESCRIPTION
**Pull request by tadamski**
https://issues.redhat.com/browse/WFLY-14682

This is an IJ upgrade that implements:
https://issues.redhat.com/browse/WFLY-14347 (https://issues.redhat.com/browse/EAP7-1189)
+ additional performance fixes to: https://issues.redhat.com/browse/EAP7-168

### CODE REVIEW

File: connector/src/main/java/org/jboss/as/connector/util/ModelNodeUtil.java [hunk 1]
```diff
@@ -41,16 +65,36 @@ public static String getResolvedStringIfSetOrGetDefault(final OperationContext c
...
+                moduleName = getResolvedStringIfSetOrGetDefault(operationContext, dataSourceNode, moduleNameAttribute);
+            }
+
+            ClassLoader moduleClassLoader = null;
+            if (moduleName != null) {
+                try {
+                    final ModuleIdentifier exceptionSorterModuleId = ModuleIdentifier.create(moduleName);
+                    moduleClassLoader = Module.getCallerModuleLoader().loadModule(exceptionSorterModuleId).getClassLoader();
+                } catch (ModuleLoadException exception) {
+                    throw new OperationFailedException("Wrong module name", exception);
```
Comment (bstansberry) [inline connector/src/main/java/org/jboss/as/connector/util/ModelNodeUtil.java:90]: This needs i18n

File: testsuite/integration/smoke/src/test/config/arq/arquillian.xml [hunk 2]
```diff
@@ -7,7 +7,7 @@
     <container qualifier="jboss" default="true">
         <configuration>
             <property name="jbossHome">${jboss.install.dir}</property>
-            <property name="javaVmArguments">${server.jvm.args} -Djboss.inst=${basedir}/target/wildfly</property>
+	    <property name="javaVmArguments">${server.jvm.args} -Djboss.inst=${basedir}/target/wildfly</property>
```
Comment (bstansberry) [inline testsuite/integration/smoke/src/test/config/arq/arquillian.xml:10]: Minor: looks like this change should be removed.

**Review by bstansberry — CHANGES_REQUESTED** [review]
The one commit here is a management API change so the subsystem version needs to be incremented, transformers are needed, and the new attributes should be covered in the subsystem test.

Note that there are 3 different things in this PR, and one is a Feature Request that requires the kanban process.  The others are a micro component upgrade and an error reporting enhancement, neither of which require kanban.  So if you want the non-RFE commits merged faster you can file a separate PR with just those two.

File: connector/src/main/java/org/jboss/as/connector/util/ModelNodeUtil.java (hunk 1 above)
Comment (tadamski) [reply connector/src/main/java/org/jboss/as/connector/util/ModelNodeUtil.java]: fixed

File: testsuite/integration/smoke/src/test/config/arq/arquillian.xml (hunk 2 above)
Comment (tadamski) [reply testsuite/integration/smoke/src/test/config/arq/arquillian.xml]: fixed

**Comment by tadamski** [comment]
@bstansberry  I decided to combine those two cause I wanted to avoid doing IronJacamar release for each JIRA and with recent IJ upgrade those are needed to compile. Anyway, this should be fixed now.

**Comment by tadamski** [comment]
hi @bstansberry, I received comment from Carlo that EAP7-1189 API change "violates backwards compatibility requirement for micro releases". Should this change be reverted?

**Comment by bstansberry** [comment]
@tadamski If a change in an IJ maintenance branch introduced some kind of incompatibility, yes, it should be reverted.

If the incompatible change is in some kind user-accessible it's best to avoid doing it, even in a new minor.

**Comment by tadamski** [comment]
@bstansberry I have reverted the API  modifing change and removed the associated commit from this PR

**Comment by bstansberry** [comment]
Thanks @tadamski. This is still held up by needing to go through the RFE process for the WFLY-14347 commit though.

**Comment by darranl** [comment]
@tadamski what is the current status of this one?  Are we still looking to get this through the RFE process?

**Comment by tadamski** [comment]
@darranl I think this should be closed, it is going through RFE process and was replaced by:
https://github.com/wildfly/wildfly/pull/14598

**Comment by bstansberry** [comment]
@tadamski If the WFLY-14313 fix is still needed can you break that commit out into a separate PR, and then close this one please? Thanks.

**Comment by tadamski** [comment]
@bstansberry WFLY-14313 was already included as a part of another PR as well, I'm closing this one

**Pull request closed without merging**